capycli/bom/create_components.py

```diff
@@ -363,15 +361,15 @@ def upload_file(
             filename = str(CycloneDxSupport.get_ext_ref_source_file(cx_comp))
             filehash = str(CycloneDxSupport.get_source_file_hash(cx_comp))
 
+            if filename is not None and filename.endswith('.git'):
+                print_red("    WARNING: resetting filename to prevent uploading .git file")
+                filename = None
+
```

Comment:
As your testcase pointed me to (see my other comment), this code actually doesn't lead to CaPyCli skipping the upload. You probably have to move your code down after lines 376-381 which re-fill the filename you invalidated:

```python
    if (filename is None or filename == "") and url:
            filename_parsed = urlparse(url)
            if filename_parsed:
                filename = os.path.basename(filename_parsed.path)
```

Reply:
You are right. I've addressed this my shifting after this check and corrected the test.